Android build error with latest version

### Description

Upgraded from 3.14.0 to 3.15.4 and cannot build in android anymore.

```

> Task :react-native-reanimated:buildCMakeDebug[arm64-v8a] FAILED
w: Detected multiple Kotlin daemon sessions at kotlin/sessions

FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':react-native-reanimated:buildCMakeDebug[arm64-v8a]'.
> com.android.ide.common.process.ProcessException: ninja: Entering directory `/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/android/.cxx/Debug/2w235572/arm64-v8a'
  [0/2] Re-checking globbed directories...
  [1/38] Building CXX object src/main/worklets/CMakeFiles/worklets.dir/__/cpp/PlatformLogger.cpp.o
  [2/38] Building CXX object src/main/reanimated/CMakeFiles/reanimated.dir/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/reanimated/Fabric/PropsRegistry.cpp.o
  [3/38] Building CXX object src/main/reanimated/CMakeFiles/reanimated.dir/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/reanimated/Fabric/ReanimatedCommitHook.cpp.o
  [4/38] Building CXX object src/main/reanimated/CMakeFiles/reanimated.dir/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/reanimated/Fabric/ReanimatedMountHook.cpp.o
  [5/38] Building CXX object src/main/reanimated/CMakeFiles/reanimated.dir/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/reanimated/Fabric/ShadowTreeCloner.cpp.o
  [6/38] Building CXX object src/main/worklets/CMakeFiles/worklets.dir/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/worklets/Registries/WorkletRuntimeRegistry.cpp.o
  [7/38] Building CXX object src/main/reanimated/CMakeFiles/reanimated.dir/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/reanimated/LayoutAnimations/LayoutAnimationsProxy.cpp.o
  [8/38] Building CXX object src/main/reanimated/CMakeFiles/reanimated.dir/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/reanimated/LayoutAnimations/LayoutAnimationsUtils.cpp.o
  [9/38] Building CXX object src/main/worklets/CMakeFiles/worklets.dir/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/worklets/Tools/UIScheduler.cpp.o
  [10/38] Building CXX object src/main/worklets/CMakeFiles/worklets.dir/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/worklets/Tools/ReanimatedJSIUtils.cpp.o
  [11/38] Building CXX object src/main/worklets/CMakeFiles/worklets.dir/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/worklets/Tools/JSLogger.cpp.o
  [12/38] Building CXX object src/main/worklets/CMakeFiles/worklets.dir/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/worklets/Tools/AsyncQueue.cpp.o
  [13/38] Building CXX object src/main/worklets/CMakeFiles/worklets.dir/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/worklets/Tools/JSScheduler.cpp.o
  [14/38] Building CXX object src/main/worklets/CMakeFiles/worklets.dir/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/worklets/Tools/WorkletEventHandler.cpp.o
  [15/38] Building CXX object src/main/worklets/CMakeFiles/worklets.dir/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/worklets/Tools/JSISerializer.cpp.o
  [16/38] Building CXX object src/main/worklets/CMakeFiles/worklets.dir/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/worklets/WorkletRuntime/ReanimatedRuntime.cpp.o
  [17/38] Building CXX object src/main/reanimated/CMakeFiles/reanimated.dir/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/reanimated/Tools/FeaturesConfig.cpp.o
  [18/38] Building CXX object src/main/reanimated/CMakeFiles/reanimated.dir/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/reanimated/AnimatedSensor/AnimatedSensorModule.cpp.o
  [19/38] Building CXX object src/main/worklets/CMakeFiles/worklets.dir/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/worklets/WorkletRuntime/RNRuntimeDecorator.cpp.o
  [20/38] Building CXX object src/main/reanimated/CMakeFiles/reanimated.dir/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/reanimated/NativeModules/NativeReanimatedModuleSpec.cpp.o
  [21/38] Building CXX object src/main/worklets/CMakeFiles/worklets.dir/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/worklets/WorkletRuntime/ReanimatedHermesRuntime.cpp.o
  [22/38] Building CXX object src/main/worklets/CMakeFiles/worklets.dir/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/worklets/Registries/EventHandlerRegistry.cpp.o
  [23/38] Building CXX object src/main/reanimated/CMakeFiles/reanimated.dir/__/cpp/PlatformLogger.cpp.o
  [24/38] Building CXX object src/main/reanimated/CMakeFiles/reanimated.dir/__/cpp/TurboModule.cpp.o
  [25/38] Building CXX object src/main/worklets/CMakeFiles/worklets.dir/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/worklets/Tools/ReanimatedVersion.cpp.o
  [26/38] Building CXX object src/main/worklets/CMakeFiles/worklets.dir/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/worklets/WorkletRuntime/WorkletRuntime.cpp.o
  [27/38] Building CXX object src/main/worklets/CMakeFiles/worklets.dir/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/worklets/SharedItems/Shareables.cpp.o
  [28/38] Building CXX object src/main/reanimated/CMakeFiles/reanimated.dir/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/reanimated/LayoutAnimations/LayoutAnimationsManager.cpp.o
  [29/38] Building CXX object src/main/reanimated/CMakeFiles/reanimated.dir/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/reanimated/NativeModules/NativeReanimatedModule.cpp.o
  FAILED: src/main/reanimated/CMakeFiles/reanimated.dir/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/reanimated/NativeModules/NativeReanimatedModule.cpp.o 
  /Users/bayram/Library/Android/sdk/ndk/23.1.7779620/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang++ --target=aarch64-none-linux-android26 --sysroot=/Users/bayram/Library/Android/sdk/ndk/23.1.7779620/toolchains/llvm/prebuilt/darwin-x86_64/sysroot -Dreanimated_EXPORTS -I/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/android/../Common/cpp/reanimated/AnimatedSensor -I/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/android/../Common/cpp/reanimated/Fabric -I/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/android/../Common/cpp/reanimated/LayoutAnimations -I/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/android/../Common/cpp/reanimated/NativeModules -I/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/android/../Common/cpp/reanimated/Tools -I/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/android/src/main/cpp -I/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native/ReactCommon -I/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native/ReactAndroid/src/main/jni/react/turbomodule -I/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native/ReactCommon/callinvoker -I/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native/ReactCommon/runtimeexecutor -I/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/android/../Common/cpp/worklets/Registries -I/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/android/../Common/cpp/worklets/SharedItems -I/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/android/../Common/cpp/worklets/Tools -I/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/android/../Common/cpp/worklets/WorkletRuntime -isystem /Users/bayram/.gradle/caches/transforms-3/37ca6e71cd1042da9bb8ace1febb9fc2/transformed/jetified-react-android-0.72.14-debug/prefab/modules/folly_runtime/include -isystem /Users/bayram/.gradle/caches/transforms-3/37ca6e71cd1042da9bb8ace1febb9fc2/transformed/jetified-react-android-0.72.14-debug/prefab/modules/glog/include -isystem /Users/bayram/.gradle/caches/transforms-3/37ca6e71cd1042da9bb8ace1febb9fc2/transformed/jetified-react-android-0.72.14-debug/prefab/modules/jsi/include -isystem /Users/bayram/.gradle/caches/transforms-3/37ca6e71cd1042da9bb8ace1febb9fc2/transformed/jetified-react-android-0.72.14-debug/prefab/modules/reactnativejni/include -isystem /Users/bayram/.gradle/caches/transforms-3/cf752b49e9002612d7c057f80c5fa6c7/transformed/jetified-fbjni-0.3.0/prefab/modules/fbjni/include -isystem /Users/bayram/.gradle/caches/transforms-3/799e00004beb768bd73045357f192d32/transformed/jetified-hermes-android-0.72.14-debug/prefab/modules/libhermes/include -isystem /Users/bayram/.gradle/caches/transforms-3/37ca6e71cd1042da9bb8ace1febb9fc2/transformed/jetified-react-android-0.72.14-debug/prefab/modules/hermes_executor/include -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fexceptions -frtti -stdlib=libc++ -DREACT_NATIVE_MINOR_VERSION=72 -DREANIMATED_VERSION=3.15.4 -DHERMES_ENABLE_DEBUGGER=1 -fexceptions -fno-omit-frame-pointer -frtti -fstack-protector-all -std=c++17 -Wall -Werror -DJS_RUNTIME_HERMES=1 -g  -fno-limit-debug-info -fPIC -DFOLLY_NO_CONFIG=1 -DFOLLY_HAVE_CLOCK_GETTIME=1 -DFOLLY_USE_LIBCPP=1 -DFOLLY_MOBILE=1 -DFOLLY_HAVE_RECVMMSG=1 -DFOLLY_HAVE_PTHREAD=1 -DFOLLY_HAVE_XSI_STRERROR_R=1 -std=gnu++17 -MD -MT src/main/reanimated/CMakeFiles/reanimated.dir/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/reanimated/NativeModules/NativeReanimatedModule.cpp.o -MF src/main/reanimated/CMakeFiles/reanimated.dir/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/reanimated/NativeModules/NativeReanimatedModule.cpp.o.d -o src/main/reanimated/CMakeFiles/reanimated.dir/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/reanimated/NativeModules/NativeReanimatedModule.cpp.o -c /Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/reanimated/NativeModules/NativeReanimatedModule.cpp
  /Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/reanimated/NativeModules/NativeReanimatedModule.cpp:210:30: error: explicit capture of 'this' with a capture default of '=' is a C++20 extension [-Werror,-Wc++20-extensions]
    uiScheduler_->scheduleOnUI(COPY_CAPTURE_WITH_THIS {
                               ^
  /Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/reanimated/NativeModules/NativeReanimatedModule.cpp:39:37: note: expanded from macro 'COPY_CAPTURE_WITH_THIS'
  #define COPY_CAPTURE_WITH_THIS [ =, this ] // NOLINT (whitespace/braces)
                                      ^
  /Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/reanimated/NativeModules/NativeReanimatedModule.cpp:276:30: error: explicit capture of 'this' with a capture default of '=' is a C++20 extension [-Werror,-Wc++20-extensions]
    uiScheduler_->scheduleOnUI(COPY_CAPTURE_WITH_THIS {
                               ^
  /Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/reanimated/NativeModules/NativeReanimatedModule.cpp:39:37: note: expanded from macro 'COPY_CAPTURE_WITH_THIS'
  #define COPY_CAPTURE_WITH_THIS [ =, this ] // NOLINT (whitespace/braces)
                                      ^
  /Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/reanimated/NativeModules/NativeReanimatedModule.cpp:290:7: error: explicit capture of 'this' with a capture default of '=' is a C++20 extension [-Werror,-Wc++20-extensions]
        COPY_CAPTURE_WITH_THIS
        ^
  /Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/reanimated/NativeModules/NativeReanimatedModule.cpp:39:37: note: expanded from macro 'COPY_CAPTURE_WITH_THIS'
  #define COPY_CAPTURE_WITH_THIS [ =, this ] // NOLINT (whitespace/braces)
                                      ^
  /Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/reanimated/NativeModules/NativeReanimatedModule.cpp:389:7: error: explicit capture of 'this' with a capture default of '=' is a C++20 extension [-Werror,-Wc++20-extensions]
        COPY_CAPTURE_WITH_THIS
        ^
  /Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/reanimated/NativeModules/NativeReanimatedModule.cpp:39:37: note: expanded from macro 'COPY_CAPTURE_WITH_THIS'
  #define COPY_CAPTURE_WITH_THIS [ =, this ] // NOLINT (whitespace/braces)
                                      ^
  /Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/reanimated/NativeModules/NativeReanimatedModule.cpp:892:7: error: explicit capture of 'this' with a capture default of '=' is a C++20 extension [-Werror,-Wc++20-extensions]
        COPY_CAPTURE_WITH_THIS
        ^
  /Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/reanimated/NativeModules/NativeReanimatedModule.cpp:39:37: note: expanded from macro 'COPY_CAPTURE_WITH_THIS'
  #define COPY_CAPTURE_WITH_THIS [ =, this ] // NOLINT (whitespace/braces)
                                      ^
  5 errors generated.
  [30/38] Building CXX object src/main/worklets/CMakeFiles/worklets.dir/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/worklets/WorkletRuntime/WorkletRuntimeDecorator.cpp.o
  [31/38] Building CXX object src/main/reanimated/CMakeFiles/reanimated.dir/__/cpp/AndroidUIScheduler.cpp.o
  [32/38] Building CXX object src/main/reanimated/CMakeFiles/reanimated.dir/__/cpp/LayoutAnimations.cpp.o
  [33/38] Building CXX object src/main/reanimated/CMakeFiles/reanimated.dir/__/cpp/JNIHelper.cpp.o
  [34/38] Building CXX object src/main/reanimated/CMakeFiles/reanimated.dir/__/cpp/OnLoad.cpp.o
  [35/38] Building CXX object src/main/reanimated/CMakeFiles/reanimated.dir/Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/Common/cpp/reanimated/Tools/UIRuntimeDecorator.cpp.o
  [36/38] Building CXX object src/main/reanimated/CMakeFiles/reanimated.dir/__/cpp/NativeProxy.cpp.o
  ninja: build stopped: subcommand failed.
  
  C++ build system [build] failed while executing:
      /Users/bayram/Library/Android/sdk/cmake/3.22.1/bin/ninja \
        -C \
        /Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/android/.cxx/Debug/2w235572/arm64-v8a \
        reanimated \
        worklets
    from /Users/bayram/Desktop/Sorsware/omuzreact/node_modules/react-native-reanimated/android

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
java.lang.StackOverflowError (no error message)

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.5.1/userguide/command_line_interface.html#sec:command_line_warnings

BUİLD FAILED in 2m 29s

```

### Steps to reproduce

1. ./gradlew clean (in mac)
2. ./gradlew assembleDebug (to generate debug apk)
3.  crashes 

### Snack or a link to a repository

-

### Reanimated version

3.15.4

### React Native version

0.72.14

### Platforms

Android

### JavaScript runtime

Hermes

### Workflow

React Native

### Architecture

Fabric (New Architecture)

### Build type

Debug app & dev bundle

### Device

None

### Device model

_No response_

### Acknowledgements

Yes

--- a/packages/react-native-reanimated/Common/cpp/reanimated/NativeModules/NativeReanimatedModule.cpp
+++ b/packages/react-native-reanimated/Common/cpp/reanimated/NativeModules/NativeReanimatedModule.cpp
@@ -36,7 +36,7 @@
 
 // Standard `__cplusplus` macro reference:
 // https://en.cppreference.com/w/cpp/preprocessor/replace#Predefined_macros
-#if REACT_NATIVE_MINOR_VERSION >= 75 || __cplusplus >= 20202L
+#if REACT_NATIVE_MINOR_VERSION >= 75 || __cplusplus >= 202002L
 // Implicit copy capture of `this` is deprecated in NDK27, which uses C++20.
 #define COPY_CAPTURE_WITH_THIS [ =, this ] // NOLINT (whitespace/braces)
 #else
@@ -44,7 +44,7 @@
 // explicitly disallows C++20 features, including the syntax above. Therefore we
 // fallback to the deprecated syntax here.
 #define COPY_CAPTURE_WITH_THIS [=] // NOLINT (whitespace/braces)
-#endif // REACT_NATIVE_MINOR_VERSION >= 75 || __cplusplus >= 20202L
+#endif // REACT_NATIVE_MINOR_VERSION >= 75 || __cplusplus >= 202002L
 
 using namespace facebook;
 

